tests/test_utils/test_linalg/test_orthogonalize.py

```diff
@@ -0,0 +1,79 @@
+"""Tests for orthogonalization functions."""
+
+from typing import Callable, Union
+
+import numpy as np
+import pytest
+from pytest_cases import parametrize
+
+from probnum import linops
+from probnum.utils.linalg import (
+    double_gram_schmidt,
+    gram_schmidt,
+    modified_gram_schmidt,
+)
+
+rng = np.random.default_rng(42)
+n = 100
+
+
+@parametrize("vector", rng.normal(size=(3, n)))
+@parametrize("basis_size", [1, 10, 50])
+@parametrize(
+    "inprod",
+    [
+        np.eye(n),
+        linops.Identity(n),
+        linops.Scaling(factors=1.0, shape=(n, n)),
+        np.inner,
+    ],
+)
+@parametrize(
+    "orthogonalization_fn", [gram_schmidt, modified_gram_schmidt, double_gram_schmidt]
+)
+def test_is_orthogonal(
+    vector: np.ndarray,
+    basis_size: int,
+    inprod: Union[
+        np.ndarray,
+        linops.LinearOperator,
+        Callable[[np.ndarray, np.ndarray], np.ndarray],
+    ],
+    orthogonalization_fn: Callable[[np.ndarray, np.ndarray], np.ndarray],
+):
+    # Compute orthogonal basis
+    orthogonal_basis, _ = np.linalg.qr(rng.normal(size=(vector.shape[0], basis_size)))
+    orthogonal_basis = orthogonal_basis.T
+
+    # Orthogonalize vector
+    ortho_vector = orthogonalization_fn(
+        v=vector, orthogonal_basis=orthogonal_basis, inprod=inprod
+    )
+    np.testing.assert_allclose(
+        orthogonal_basis @ ortho_vector,
+        np.zeros((basis_size,)),
+        atol=1e-12,
+        rtol=1e-12,
+    )
+
```

Comment:
`gram_schmidt` and `modified_gram_schmidt` may fail this test even if the methods are implemented correctly. This is because the accuracy of both methods depends on the condition number of the matrix being orthogonalized. However, the accuracy of `double_gram_schmidt` does not depend on the condition number of the matrix, and if it passes, this should indicate that `gram_schmidt` and `modified_gram_schmidt` are implemented correctly